src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/InMemoryPartitionedAppTokenCacheAccessor.cs

```diff
@@ -57,12 +63,26 @@ internal class InMemoryPartitionedAppTokenCacheAccessor : ITokenCacheAccessor
         #region Add
         public void SaveAccessToken(MsalAccessTokenCacheItem item)
         {
+            if (IsCacheOverCapacity(AccessTokenSizeInBytes))
+            {
+                _logger.Always("[AppCache] Cache is over capacity.");
+                Compact();
+            }
+
             string itemKey = item.GetKey().ToString();
             string partitionKey = CacheKeyFactory.GetClientCredentialKey(item.ClientId, item.TenantId, item.KeyId);
 
+            // Update cache size only if cache item is added, not updated
+            if (!AccessTokenCacheDictionary.TryGetValue(partitionKey, out var partition) || !partition.TryGetValue(itemKey, out _))
+            {
+                Interlocked.Add(ref _appCacheSize, AccessTokenSizeInBytes);
```

Comment:
>_appCacheSize [](http://example.com/codeflow?start=36&length=13)

Is it really necessary to use the variable to track the sizes? Isn't _appCacheSize == AccessTokenCacheDictionary.Count * AccessTokenSizeInBytes?

There seems to be some risks of not setting the sizes correctly even Interlock() is atomic. And it is more complicated to set them in the Clear() method without locking.

Reply:
`ConcurrentDictionary<string, ConcurrentDictionary<string, MsalAccessTokenCacheItem>> AccessTokenCacheDictionary`
TokenCacheDictionaries are dictionary of dictionaries. with the outer key being a partition key and an inner key is an actual cache item key. So unfortunately count does not work.

Reply:
If AccessTokenSizeInBytes is added to the _appCacheSize every time a token is saved, isn't _appCacheSize == AccessTokenCacheDictionary.Count * AccessTokenSizeInBytes?